app/__init__.py

```diff
@@ -34,106 +40,197 @@
     }
 )
 
+# path navigating to current workspace directory
+WORKING_PATH = os.path.normpath(app.root_path + WORKING_DIR + r' ')
+
+
+def get_connection():
+    """
+        Method to create an object of subclass and create a connection
+        :return : an instantiated object for class RestDB
+    """
+    return RestDB()
+
 # Route for creating workspaces
 @app.route('/workspace/create', methods=['POST'])
-def linchpin_init():
+def linchpin_init() -> Response:
+    """
+        POST request route for creating workspaces.
+        RequestBody: {"name": "workspacename"}
+        :return : response with created workspace name,
+                  status and code
+    """
     try:
-        data = request.json     # Get request body
+        data = request.json  # Get request body
         name = data["name"]
-        # Checking if workspace already exists
-        if os.path.exists(WORKING_DIR + "/" + name):
-            return jsonify(status="Workspace already exists")
-        else:
-            output = subprocess.Popen(["linchpin", "-w " +
-                                       WORKING_DIR + name +
-                                       "/", "init"], stdout=subprocess.PIPE)
-            return jsonify(name=data["name"],
-                           status="Workspace created successfully",
-                           Code=output.returncode)
-    except Exception as e:
-        app.logger.error(e)
-        return jsonify(status=409, code=output.returncode)
+        identity = str(uuid.uuid4()) + "_" + name
+        try:
+            get_connection().db_insert(identity, name,
+                                       response.WORKSPACE_REQUESTED)
+            if not re.match("^[a-zA-Z0-9]*$", name):
+                get_connection().db_update(identity, response.WORKSPACE_FAILED)
+                return jsonify(status=errors.ERROR_STATUS,
+                               message=errors.INVALID_NAME)
+            else:
+                # Checking if workspace already exists
+                output = subprocess.Popen(["linchpin", "-w " +
+                                          WORKING_DIR + identity +
+                                          "/", "init"], stdout=subprocess.PIPE)
+                get_connection().db_update(identity, response.WORKSPACE_SUCCESS)
+                return jsonify(name=data["name"], id=identity,
+                               status=response.CREATE_SUCCESS,
+                               Code=output.returncode,
+                               mimetype='application/json')
+        except Exception as e:
+            get_connection().db_update(identity, response.WORKSPACE_FAILED)
+            app.logger.error(e)
+            return jsonify(status=errors.ERROR_STATUS, message=str(e))
+    except (KeyError, ValueError, TypeError):
```

Comment:
@mansikulkarni96 Why do we need to have multiple try blocks? 
cant we handle both exceptions by single try except block in this function??

Reply:
In order to update db when the exception occurs I need the id field of workspace which becomes inaccessible in the except block below it, so I moved it outside but we cannot have no exception handling for that part as the name,key, type errors may occur, hence I have kept a nested exception handler.